lib/elixir/src/elixir_lexical.erl

```diff
@@ -99,7 +99,17 @@ warn_unused_imports(Pid, E) ->
   ok.
 
 warn_unused_requires(Pid, E) ->
-  [elixir_errors:file_warn(Meta, ?key(E, file), ?MODULE, {unused_require, Module})
+  [case lists:keyfind(opts, 1, Meta) of
+     {opts, Opts} ->
+       case lists:keyfind(as, 1, Opts) of
+         {as, Alias} when is_atom(Alias) ->
+           elixir_errors:file_warn(Meta, ?key(E, file), ?MODULE, {unused_require, Module, Alias});
```

Comment:
What will the warning be if the alias is also not used? Will it trigger the regular unused alias warning?